src/libbson/src/bson/bson.c

```diff
@@ -3236,9 +3325,12 @@ bson_array_as_json (const bson_t *bson, size_t *length)
    state.depth = 0;
    state.err_offset = &err_offset;
    state.mode = BSON_JSON_MODE_LEGACY;
+   state.max_len = BSON_MAX_LEN_UNLIMITED; // TODO: how does the limit get here?
```

Comment:
I think we would need a `bson_array_as_json_with_opts` for callers to pass a limit, right?

Reply:
Yes, that's correct.  Do we want to add this?  Or leave it as-is for now, with no limits on arrays?